### PR TITLE
[hotfix] cover non-tested data table v4 code path

### PR DESCRIPTION
**Pull request by walterddr**
At first I thought datatable version has been defaulted to v4 when committing #9731 . but apparently it is not even tested.

created this hotfix follow up as well as adding abundant tests.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/datatable/DataTableImplV4.java [hunk 1]
```diff
@@ -62,12 +62,23 @@ public class DataTableImplV4 extends BaseDataTable {
   protected byte[] _variableSizeDataBytes;
   protected ByteBuffer _variableSizeData;
   protected Map<String, String> _metadata;
-  private static final int VERSION = 1;
   protected int[] _columnOffsets;
   protected int _rowSizeInBytes;
 
   public DataTableImplV4() {
-    super();
+    _numRows = 0;
```
Comment (agavra) [inline pinot-common/src/main/java/org/apache/pinot/common/datatable/DataTableImplV4.java:69]: not sure I can tell from just reviewing the PR, what was the bug?

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/datatable/DataTableImplV4.java:69]: see my previous comment :-)

**Comment by siddharthteotia** [comment]
Since this is labelled as hotfix, I am wondering if you have started using V4. 

AFAIK, it wasn't tested and there was plan to move over to V4 since it also has NULL handling support IIRC. The current / default version is V3.

**Review by siddharthteotia — APPROVED** [review]

**Comment by walterddr** [comment]
> Since this is labelled as hotfix, I am wondering if you have started using V4.
> 
> AFAIK, it wasn't tested and there was plan to move over to V4 since it also has NULL handling support IIRC. The current / default version is V3.

label as hotfix because this was a regression introduced a bit earlier: 
- when i decouple the code in #9731 , the DataTableImplV4 becomes a copy of the original BaseDataBlock, but since it is no longer a data block, we dont have the test coverage via the DataBlockTest, and neither was it tested in the DataTableTest b/c it doesn't test V4 (only V3)
- it is not being used after the decoupling, but it is still good to test the V4 impl.

**Pull request merged**